aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -862,6 +862,23 @@ Tensor max(const Tensor& self, const Tensor& other) {
   return at::maximum(self, other);
 }
 
+Tensor& fmax_out(Tensor& result, const Tensor& self, const Tensor& other) {
+    auto self_isnan = self.isnan();
```

Comment:
Include a fast-path for integer dtypes. Since integer cannot be nan, you can simply return at::gt_out(result, self, other).

Reply:
at::maximum_out, not at::gt_out, though

Reply:
But see my latest comment below about switching to use TensorIterator for simplicity and performance.

Reply:
I've improved the code to use TensorIterator instead.